### PR TITLE
app-misc/g810-led: bump to 0.4.2 + various fixes

### PR DESCRIPTION
**Pull request by crabbedhaloablution**
0.4.2 brings support for g815, builds with clang, and various minor fixes.

Ebuild fixes:
Create and populate /etc/g810-led/ as in upstream makefile so systemd
.services work.
Change udev rules to use uaccess instead of MODE="666".
Set CXX to system compiler
Build and use "bin-linked" g810-led. Same features, less space.
Include headers in install
Add pkg_postinst with hints about use

Closes: https://bugs.gentoo.org/696468
Closes: https://bugs.gentoo.org/729288
Package-Manager: Portage-3.0.12, Repoman-3.0.2
Signed-off-by: Peter Alfredsen <crabbedhaloablution@icloud.com>

### CODE REVIEW

File: app-misc/g810-led/g810-led-0.4.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	emake LIB="$(usex hidapi hidapi libusb)" CXX="$(tc-getCXX)" bin-linked
+}
+
+src_install() {
+	dolib.so "lib/libg810-led.so.${PV}"
+	dosym "libg810-led.so.${PV}" "/usr/$(get_libdir)/libg810-led.so"
+
+	insinto /etc/g810-led/
+	newins sample_profiles/group_keys profile
+	newins sample_profiles/all_off reboot
```
Comment (lanodan) [inline app-misc/g810-led/g810-led-0.4.2.ebuild:46]: Those are installed as documentation, so you should either remove this new block or remove `sample_profiles` from `DOCS`.

File: app-misc/g810-led/g810-led-0.4.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+	dobin bin/g810-led
+	local boards=(213 410 413 512 513 610 815 910 pro)
+	local x
+	for x in "${boards[@]}"; do
+		dosym g810-led "/usr/bin/g${x}-led"
+	done
+
+	insinto /usr/include/g810-led
+	doins src/classes/*.h
```
Comment (lanodan) [inline app-misc/g810-led/g810-led-0.4.2.ebuild:56]: Just pure curiosity to see if I should fix 0.3.6: Is there other programs out there using g810-led as a library?

**Review by lanodan — CHANGES_REQUESTED** [review]
Thanks for this version bump, I'm still using the 0.3.6 version and didn't try the 0.4.x branch but I guess it still breaks parts of my keyboard so that old version should be kept for now.

File: app-misc/g810-led/g810-led-0.4.2.ebuild (hunk 2 above)
Comment (crabbedhaloablution) [reply app-misc/g810-led/g810-led-0.4.2.ebuild:56]: Not to my knowledge

File: app-misc/g810-led/g810-led-0.4.2.ebuild (hunk 1 above)
Comment (crabbedhaloablution) [reply app-misc/g810-led/g810-led-0.4.2.ebuild:46]: I disagree.
The intention is that the files installed in /etc should be edited.
The sample_profiles provide examples of how to make a custom profile. Both are needed.

Comment (juippis) [reply app-misc/g810-led/g810-led-0.4.2.ebuild:46]: @lanodan what do you think? LGTM.

Comment (lanodan) [reply app-misc/g810-led/g810-led-0.4.2.ebuild:46]: Well, yeah, why not. LGTM

**Pull request closed without merging**